pkg/discovery/kv/kv.go

```diff
@@ -63,9 +63,23 @@ func (s *Discovery) Initialize(uris string, heartbeat time.Duration, ttl time.Du
 	s.ttl = ttl
 	s.path = path.Join(s.prefix, discoveryPath)
 
+	var config *store.Config
```

Comment:
@dhiltgen So what's happening now is that I really delegate the work of initializing TLS to the etcd client. For Consul it's different, I have some control over what I give to the `HttpTransport`. So this is something that really is on the libkv side. I don't think this impacts this PR in some way.